experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,399 @@
+	return e.pairs
+}
+
+type bridgeFieldEncoder struct {
+	pairs []otelcore.KeyValue
+}
+
+var _ otlog.Encoder = &bridgeFieldEncoder{}
+
+func (e *bridgeFieldEncoder) EmitString(key, value string) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitBool(key string, value bool) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitInt(key string, value int) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitInt32(key string, value int32) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitInt64(key string, value int64) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitUint32(key string, value uint32) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitUint64(key string, value uint64) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitFloat32(key string, value float32) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitFloat64(key string, value float64) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitObject(key string, value interface{}) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitLazyLogger(value otlog.LazyLogger) {
+	value(e)
+}
+
+func (e *bridgeFieldEncoder) emitCommon(key string, value interface{}) {
+	e.pairs = append(e.pairs, otTagToOtelCoreKeyValue(key, value))
+}
+
+func otLogFieldsToOtelCoreKeyValues(fields []otlog.Field) []otelcore.KeyValue {
+	encoder := &bridgeFieldEncoder{}
+	for _, field := range fields {
+		field.Marshal(encoder)
+	}
+	return encoder.pairs
+}
+
+func (s *bridgeSpan) LogKV(alternatingKeyValues ...interface{}) {
+	fields, err := otlog.InterleavedKVToFields(alternatingKeyValues...)
+	if err != nil {
+		return
+	}
+	s.LogFields(fields...)
+}
+
+func (s *bridgeSpan) SetBaggageItem(restrictedKey, value string) ot.Span {
+	s.ctx.setBaggageItem(restrictedKey, value)
+	return s
+}
+
+func (s *bridgeSpan) BaggageItem(restrictedKey string) string {
+	return s.ctx.baggageItem(restrictedKey)
+}
+
+func (s *bridgeSpan) Tracer() ot.Tracer {
+	return s.tracer
+}
+
+func (s *bridgeSpan) LogEvent(event string) {
+	s.LogEventWithPayload(event, nil)
+}
+
+func (s *bridgeSpan) LogEventWithPayload(event string, payload interface{}) {
+	data := ot.LogData{
+		Event:   event,
+		Payload: payload,
+	}
+	s.Log(data)
+}
+
+func (s *bridgeSpan) Log(data ot.LogData) {
+	record := data.ToLogRecord()
+	s.LogFields(record.Fields...)
+}
+
+type bridgeTracer struct {
+	otelTracer oteltrace.Tracer
+}
+
+var _ ot.Tracer = &bridgeTracer{}
+
+func NewTracer(tracer oteltrace.Tracer) ot.Tracer {
+	return &bridgeTracer{
+		otelTracer: tracer,
+	}
+}
+
+func NewTracerFromGlobal() ot.Tracer {
+	return NewTracer(oteltrace.GlobalTracer())
+}
+
+func (t *bridgeTracer) StartSpan(operationName string, opts ...ot.StartSpanOption) ot.Span {
+	sso := ot.StartSpanOptions{}
+	for _, opt := range opts {
+		opt.Apply(&sso)
+	}
+	// TODO: handle links, needs SpanData to be in the API first?
```

Comment:
`SpanData` is being removed in favor of language-appropriate start options to convey this information. There are some questions about links, see https://github.com/open-telemetry/opentelemetry-go/pull/80